src/System.CommandLine/Parsing/CommandArgumentNode.cs

```diff
@@ -10,10 +12,7 @@ internal class CommandArgumentNode : SyntaxNode
             Argument argument,
             CommandNode parent) : base(token, parent)
         {
-            if (token.Type != TokenType.Argument)
-            {
-                throw new ArgumentException($"Incorrect token type: {token}");
-            }
+            Debug.Assert(token.Type == TokenType.Argument, $"Incorrect token type: {token}");
```

Comment:
these types are internal and we control the input, so there is no need to the throw condition. `Debug.Assert` will keep us safe (the CI runs all the tests in debug) with 0 cost for the `Release` build (less IL to compile, possibility to inline the method)